src/middlewared/middlewared/plugins/catalogs_linux/update.py

```diff
@@ -22,6 +22,7 @@ class CatalogModel(sa.Model):
     repository = sa.Column(sa.Text(), nullable=False)
     branch = sa.Column(sa.String(255), nullable=False)
     builtin = sa.Column(sa.Boolean(), nullable=False, default=False)
+    preferred_trains = sa.Column(sa.JSON(type=list))
```

Comment:
Why don't we make it `nullable=False`?

Reply:
It's `nullable=False` by default

Reply:
Then your migration lacks code to set NULL column to NOT NULL